src/sailfishusermanager.cpp

```diff
@@ -465,6 +467,16 @@ int SailfishUserManager::removeUserFiles(const char *user)
     return EXIT_FAILURE;
 }
 
+/*!
+  \brief Process user session.
+ */
+int SailfishUserManager::processSession(const char *uid)
+{
+    executeScripts(QString(uid).toUInt(), USER_SESSION_SCRIPT_DIR);
```

Comment:
I think you should get the same effect if you put this call inside`onUnitJobFinished()` so that there is no need to add extra systemd units or running another copy of user-managerd with special arguments.

I suggest that you either have this call after `autologin@<uid>.service` job finishes (i.e. `autologin@.service` has started) or after `user@<uid>.service` start job finishes so it's actually a "late" service. The former would need a new branch to be added of course, e.g. `job.type == SystemdManager::StartJob && job.unit == AUTOLOGIN_SERVICE.arg(m_switchUser)`. The latter should work if you just have it after that `currentUserChanged(m_switchUser)` signal which is there already (also consider that "backup plan" in that case).

Reply:
`onUnitJobFinished` is fired only when there was an user switch operation. But here is also need to catch user session after device boot.

The service isn't running continuously, it has the exit timer, so I don't think that a `insideonUnitJobFinished` is a good idea to deal with it.

It has already used `user@<uid>.service` to launch the user-managerd to execute scripts.

Reply:
I see. That is indeed a bit different. How about not involving user-managerd at all? You could have a service user-session-scripts@.service which would contain something like this:
```ini
[Unit]
Description=Scripts on user session
Before=user@%i.service

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/libexec/user-session-scripts start %i
ExecStop=/usr/libexec/user-session-scripts stop %i
```

And then have that /usr/libexec/user-session-scripts contain a little script that would execute the scripts in those directories. Last you would have that systemd conf file so that this is executed when e.g. `autologin@<uid>.service` starts.

Reply:
I understand, but wouldn't be it better to bring together all user stuff to the one component and do not spread them among different services/scripts?

Reply:
We intended user-managerd to be a component that would take part in modifying users and in user switches because those could not be done from UI directly due to insufficient permissions but it wasn't intended to start on bootup. This change would effectively make user-managerd to start on bootup even if there are no scripts to run. I suppose something like ConditionDirectoryNotEmpty setting could prevent it in that case. Anyway since user-managerd is built on Qt it's not very lightweight service to start, which is something to consider.

We have some similar session start and end related scripts as systemd services already. They are in a different repository. The current scripting support in this repository is strictly for user creation or removal which is very much about user management, the primary use for this service, so it's in some ways different from this although maybe not too far.